streams/src/main/java/org/apache/kafka/streams/kstream/internals/AbstractKTableKTableJoinValueGetterSupplier.java

```diff
@@ -32,14 +30,15 @@ public AbstractKTableKTableJoinValueGetterSupplier(final KTableValueGetterSuppli
     public String[] storeNames() {
         final String[] storeNames1 = valueGetterSupplier1.storeNames();
         final String[] storeNames2 = valueGetterSupplier2.storeNames();
-        final ArrayList<String> stores = new ArrayList<>(storeNames1.length + storeNames2.length);
+        int idx = 0;
+        final String[] stores = new String[storeNames1.length + storeNames2.length];
         for (final String storeName : storeNames1) {
-            stores.add(storeName);
+            stores[idx++]=storeName;
         }
         for (final String storeName : storeNames2) {
-            stores.add(storeName);
+            stores[idx++]=storeName;
         }
```

Comment:
What do you think about using only one `for` on the destination `stores` array filling it in the following way ? 

`
for (int i = 0; i < stores.length; i++) {
    stores[ i ] = ( i < storeNames1.length ) ? storeNames[ i ] : storesNames2 [ i - storeNames1.length);
}
`

Reply:
I'm ok with the your suggestion. But, it introduces some conditions. Both the solutions are going to iterate the 'n' elements. It's a matter of style.

Reply:
Yeah that's true, you can just keep your way ;)

Reply:
This seems like an unnecessary micro-optimization, no?

Reply:
I'd just use `addAll` twice to make the code simpler instead of making it more complex.

Reply:
I don't want to create an intermediate list. With `addAll`, we have to convert the list back to array,

Reply:
I know the intent, I just don't know why. I don't believe the copy is a bottleneck here so it looks like premature optimization.

Reply:
Ok, then I revert this change.